Add release note for function selector resource type (#8913)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/release-notes.md

For <Constant name="fusion_engine" /> updates, refer to the [dbt-fusion changelog](https://github.com/dbt-labs/dbt-fusion/blob/main/CHANGELOG.md).

## April 2026

> - **Enhancement:** Catalog now supports `function` as a selector resource type, enabling you to filter and select [user-defined functions (UDFs)](/docs/build/udfs) using node selectors. For example, use `dbt build --select "resource_type:function"` to build all functions in your project, or `dbt list --select "resource_type:function"` to list them. This requires dbt v1.11 or later. For more information, see [Node selector methods](/reference/node-selection/methods#resource_type).

Author: ```suggestion
- **Enhancement:** Catalog now supports `function` as a resource type. You can browse, search, and filter UDFs from the sidebar and resource search, and view function details (arguments, return type, and source code) on their own detail page. Functions appear in their own mini-lineage view but are not shown in the main project DAG. For more information, refer to [user-defined functions](/docs/build/udfs).
```

- **Enhancement:** The Studio IDE now validates dbt YAML using <Constant name="fusion" /> aligned JSON Schema from [dbt-jsonschema](https://github.com/dbt-labs/dbt-jsonschema) across [dbt platform release tracks](/docs/dbt-versions/cloud-release-tracks), including for development environments on <Constant name="core" />. This improves autocomplete and structural feedback in the editor. Diagnostics can occasionally disagree with what your environment accepts; use dbt runs and previews as the source of truth. For context, see [Migrate to the latest YAML spec](/docs/build/latest-metrics-spec) and [dbt YAML validation in Studio](/docs/cloud/studio-ide/develop-in-studio#dbt-yaml-validation). This will be a phased rollout starting the week of April 6th.
- **Enhancement:** The Studio IDE status bar now offers more control, more detailed information, and quicker access to settings for deferral, dbt version, and project status. For more information, see the [Studio IDE docs](/docs/cloud/studio-ide/ide-user-interface#the-command-and-status-bar). These updates roll out in phases to existing accounts starting April 6.

> - **Enhancement:** Catalog now supports `function` as a selector resource type, enabling you to filter and select [user-defined functions (UDFs)](/docs/build/udfs) using node selectors. For example, use `dbt build --select 'resource_type:function'` to build all functions in your project, or `dbt list --select 'resource_type:function'` to list them. This requires dbt v1.11 or later. For more information, see [Node selector methods](/reference/node-selection/methods#resource_type).

Author: ```suggestion
- **Enhancement:** Catalog now supports `function` as a selector resource type, enabling you to filter and select [user-defined functions (UDFs)](/docs/build/udfs) using node selectors. For example, use `dbt build --select "resource_type:function"` to build all functions in your project, or `dbt list --select "resource_type:function"` to list them. This requires dbt v1.11 or later. For more information, see [Node selector methods](/reference/node-selection/methods#resource_type).
```

> - **Enhancement:** Catalog now supports `function` as a selector resource type, enabling you to filter and select [user-defined functions (UDFs)](/docs/build/udfs) using node selectors. For example, use `dbt build --select "resource_type:function"` to build all functions in your project, or `dbt list --select "resource_type:function"` to list them. This requires dbt v1.11 or later. For more information, see [Node selector methods](/reference/node-selection/methods#resource_type).

Reviewer: hey @runleonarun - super exciting to have this new resource type in catalog lineage!

do we want to tell users the function is now a visible node in catalog and then mix in there a dbt build command?   since the dbt build/dbt list is more of a development ide/cli command and not supported in catalog (i think). i see the following in the lineage but according to the [original pr](https://github.com/dbt-labs/metadata-ui/pull/25750), it looks like this is part of a series of changes. the function resource type still needs to be added to the catalog sidebar and selector syntax when viewing the lineage:
<img width="1004" height="708" alt="Screenshot 2026-04-08 at 09 16 08" src="https://github.com/user-attachments/assets/c96b2392-ba0c-408b-aceb-6b699bb71f8e" />
<img width="1113" height="688" alt="Screenshot 2026-04-08 at 09 23 39" src="https://github.com/user-attachments/assets/b7ae762d-b360-4eb4-8d76-1167ebae723b" />
<img width="458" height="583" alt="Screenshot 2026-04-08 at 09 25 58" src="https://github.com/user-attachments/assets/3898bbc3-5ebe-4c00-b323-e6054c420d4b" />

Reviewer: i'm not sure if it's best to wait until all the prs (4 total: 2 done and 2 outstanding) before adding this release note. what do you think? alternatively, we can add this release note to say catalog now supports the function resource type in the lineage

> - **Enhancement:** dbt Explorer now supports `function` as a selector resource type, enabling you to filter and select [user-defined functions (UDFs)](/docs/build/udfs) using node selectors. For example, use `dbt build --select "resource_type:function"` to build all functions in your project, or `dbt list --select "resource_type:function"` to list them. This requires dbt v1.11 or later. For more information, see [Node selector methods](/reference/node-selection/methods#resource_type).

Author: @copilot apply changes based on [this feedback](https://github.com/dbt-labs/docs.getdbt.com/pull/8913#discussion_r3048041148)

## April 2026

> - **Enhancement:** Catalog now supports `function` as a resource type. You can browse, search, and filter UDFs from the sidebar and resource search, and view function details (arguments, return type, and source code) on their own detail page. Functions appear in their own mini-lineage view but are not shown in the main project DAG. For more information, refer to [user-defined functions](/docs/build/udfs).

Reviewer: ```suggestion
- **Enhancement:** <Constant name="catalog"/> now supports `function` as a resource type. You can browse, search, and filter UDFs from the sidebar and resource search, and view function details (arguments, return type, and source code) on their own detail page. Functions appear in their own mini-lineage view but are not shown in the main project DAG. For more information on UDFs in dbt, refer to [user-defined functions](/docs/build/udfs).
```
